examples/targeting/search_for_language_and_carrier_constants.py

```diff
@@ -0,0 +1,171 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Demonstrates how to search for language and mobile carrier constants.
+
+Specifically, this example illustrates how to:
+1. Search for language constants where the name includes a given string.
+2. Search for all the available mobile carrier constants with a given country
+    code.
+"""
+
+
+import argparse
+import sys
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+
+def main(client, customer_id, language_name, carrier_country_code):
+    """Demonstrates how to search for language and mobile carrier constants.
+
+    Args:
+        client: An initialized Google Ads API client.
+        customer_id: The Google Ads customer ID.
+        language_name: String included in the language name to search for.
+        carrier_country_code: String code of the country where the mobile
+            carriers are located.
+    """
+    try:
+        _search_for_language_constants(client, customer_id, language_name)
+        _search_for_carrier_constants(client, customer_id, carrier_country_code)
+    except GoogleAdsException as ex:
+        print(
+            f'Request with ID "{ex.request_id}" failed with status '
+            f'"{ex.error.code().name}" and includes the following errors:'
+        )
+        for error in ex.failure.errors:
+            print(f'\tError with message "{error.message}".')
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print(f"\t\tOn field: {field_path_element.field_name}")
+        sys.exit(1)
+
+
+def _search_for_language_constants(client, customer_id, language_name):
+    """Searches for language constants where the name includes a given string.
+
+    Args:
+        client: An initialized Google Ads API client.
+        customer_id: The Google Ads customer ID.
+        language_name: String included in the language name to search for.
+    """
+    # Get the GoogleAdsService client.
+    google_ads_service = client.get_service("GoogleAdsService", version="v5")
+
+    # Create a query that retrieves the language constants where the name
+    # includes a given string.
+    query = f"""
+        SELECT language_constant.id, language_constant.code,
+        language_constant.name, language_constant.targetable
+        FROM language_constant
+        WHERE language_constant.name LIKE '%{language_name}%'"""
```

Comment:
I thought we wanted to use the SQL syntax David mentioned, so this would be formatted like:

```py
SELECT
  language_constant.id,
  language_constant.code,
  languge_constant.name,
  language_constant.targetable
FROM language_constant
WHERE
  language_constant.name LIKE '%{language_name}%'
```

Reply:
Updated both.